synapse/http/client.py

```diff
@@ -69,6 +70,21 @@ def check_against_blacklist(ip_address, ip_whitelist, ip_blacklist):
     return False
 
 
+_EPSILON = 0.00000001
+
+
+def _make_scheduler(reactor):
+    """Makes a schedular suitable for a Cooperator using the given reactor.
+
+    (This is effectively just a copy from `twisted.internet.task`)
+    """
+
+    def _scheduler(x):
+        return reactor.callLater(_EPSILON, x)
+
+    return _scheduler
```

Comment:
Looks like this is

https://github.com/twisted/twisted/blob/121c98e006a31750661107d390ec2dc4ffe28e8a/src/twisted/internet/task.py#L362-L365

So it is using the reactor given instead of the global reactor, which fixes it for tests.

Reply:
Yup. Do you want the to be in the comment or?

Reply:
I think the comment given is enough, the difference was obvious once I compared the code. I was mostly just thinking out loud here. 👍

Reply:
Cool!